docs/site/LB3-vs-LB4-request-response-cycle.md

```diff
@@ -0,0 +1,96 @@
+
+The request/response cycle infrastructure and pathway are very different in
+LoopBack 3 and LoopBack 4. Knowing the differences will help you migrate
+LoopBack 3 apps to LoopBack 4 and implement new request/response related
+features in LoopBack 4.
+
+This document will guide you through the differences and show the LoopBack 4
+equivalent, if there is any.
+
+### Request/response infrastructure
+
+The difference begins with the LoopBack application object itself. In
+LoopBack 3, it is an instance of an Express app; in LoopBack 4, it is not.
+Although LoopBack 4 uses Express as the HTTP server, it is not directly exposed
+anymore.
+
+In LoopBack 3 you could add routes and load custom middleware using `app.get()`,
+`app.post()`, `app.use()`, etc., just like how you do in Express.
+In LoopBack 4, you cannot do it anymore. The closest to doing this is to use the
+[RestApplication.mountExpressRouter()](https://loopback.io/doc/en/lb4/apidocs.rest.restapplication.mountexpressrouter.html)
+API. This API allows you to mount a custom Express router on the LoopBack app.
+
+Using [Controllers](https://loopback.io/doc/en/lb4/Controllers.html) is the
+recommended way for creating custom (and REST) endpoints on your app. Its
+support for
+[dependency injection](https://loopback.io/doc/en/lb4/Dependency-injection.html)
+and [Interceptors](https://loopback.io/doc/en/lb4/Interceptors.html) makes it a
+very powerful extension mechanism.
+
+In LoopBack 4
+[middleware.json](https://loopback.io/doc/en/lb3/middleware.json.html)
+is not require anymore because of architectural changes.
+
+In LoopBack 3, models files automatically create the corresponding REST API
+endpoints and the database query machinery (using the configured datasource).
+In LoopBack 4, model files are limited only to describing the properties of the
+data. You will have to create a corresponding
+[Repository](https://loopback.io/doc/en/lb4/Repositories.html)
+for database connectivity, and controller creating the REST API endpoint.
+
+The fact that you have to create two more artifacts along with the model to
+get a REST endpoint working might seem overly tedious at first. However, the
+separation of concerns and decoupling the functionality makes the codebase
+cleaner, easier to maintain, and much easier to customize functionality at
+various levels. This can be better appreciated as the complexity of your app
+grows.
+
+For those who are uncomfortable with the concept of having to creating a
+repository and a controller for a model, we have a component
+[@loopback/rest-crud
+](https://loopback.io/doc/en/lb4/Creating-crud-rest-apis.html)
+; with a little bit of configuration, a model file is all you will need to
+create the REST endpoints. Once your requirements outgrow what
+`@loopback/rest-crud` provides, you can implement your REST endpoints the
+idiomatic way.
+
+Components are still supported in LoopBack 4, but the concept of component
+has completely changed.
```

Comment:
I feel the components are out of scope of request/response life cycle. We have a standalone doc page to cover migration of components - see https://loopback.io/doc/en/lb4/migration-extensions.html 

Unless there is something specific to the way how components interact (or contribute to) request/response life cycle, then I am proposing to remove this and the next paragraph.

Reply:
As someone who wrote a component to add routes to the app, who is going through the docs to see what has changed in LB4 from the req-res perspective, this info will definitely help me.; instead of having to browse through the docs, with no certainty when and where I will find the info I need.